### PR TITLE
handle payload download errors gracefully

### PR DESCRIPTION
**Pull request by khyberspache**

### CODE REVIEW

File: sockets/contact.go [hunk 1]
```diff
@@ -25,18 +27,34 @@ func refreshBeacon(agent *util.AgentConfig, beacon *util.Beacon) {
...
+		}
+
+		return path, nil
+	}
+	return "", errors.New("UNHANDLED PAYLOAD EXCEPTION")
 }
 
+func payloadErrorResponse(err error, agent *util.AgentConfig, link *util.Instruction) {
+	link.Response = "Payload Error: " + strings.TrimSpace(err.Error())
+	link.Status = -1
```
Comment (privateducky) [inline sockets/contact.go:55]: why -1 and not a 1 in general to indicate not worked?

File: sockets/rawgrpc.go [hunk 2]
```diff
@@ -30,13 +30,18 @@ func (contact GRPC) Communicate(agent *util.AgentConfig, beacon util.Beacon) uti
 			}
 			for _, link := range tempB.Links {
 				var payloadPath string
+				var payloadErr error
 				if len(link.Payload) > 0 {
```
Comment (privateducky) [inline sockets/rawgrpc.go:34]: this payload block/sequence is too duplicate across the sockets. can we move to the contact.go (base) instead?

**Review by privateducky — CHANGES_REQUESTED** [review]
questions

File: sockets/rawgrpc.go (hunk 2 above)
Comment (khyberspache) [reply sockets/rawgrpc.go:34]: did a little bit of reducing, but I don't want to do too much more

**Review by privateducky — APPROVED** [review]

**Pull request merged**